### PR TITLE
EOS-15530 : LDRr2: Configure HAProxy as a part of S3 setup component provisioning

### PR DESCRIPTION
**Pull request by shahidSTX**
All HAProxy configurations are currently taken care by Provisioner.

These changes create a script that will configure the S3 part of HAProxy. As a part of it, s3_haproxy_config binary is created and executed to edit /etc/haproxy/haproxy.cfg file

### CODE REVIEW

**Comment by shahidSTX** [comment]
@nileshgovande @sachitanands Please review the _s3rpm.spec_ file changes. I have included s3haproxyconfig to do configuration of haproxy via a script and included it as a binary.

File: scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py [hunk 1]
```diff
@@ -56,7 +64,33 @@ def run(self):
...
+
+
+
+    acl s3authbackendacl dst_port 9443
+    acl s3authbackendacl dst_port 9080
+    use_backend s3-auth if s3authbackendacl
+
+
+
+# BackEnd roundrobin as balance algorithm
```
Comment (sachitanands) [inline scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:86]: are we planning for handling changes of updating all instance ids of s3server when we have multi-node/multiple instance setup.

Comment (shahidSTX) [reply scripts/haproxy/s3haproxyconfig/s3haproxyconfig/s3_haproxy_config.py:86]: Yes, that will be taken up in [EOS-16435](https://jts.seagate.com/browse/EOS-16435)

**Review by amitwac2608 — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]
looks good..

**Pull request merged**